docs/src/examples/NPG.jl

```diff
@@ -0,0 +1,171 @@
+# PPO, and other policy gradient methods.
+
+# First, let's go head and grab all the dependencies
+using LinearAlgebra, Random, Statistics # From Stdlib
+using LyceumAI         # For the NPG controller
+using LyceumMuJoCo     # For the Hopper environment
+using LyceumBase.Tools # For the ControllerIterator discussed below
+using Flux             # For our Neural Network Needs
+using Flux: glorot_uniform
+using UniversalLogger
+using Plots
+
+# We first configure and instantiate of our `Hopper` environment to grab useful
+# environment specific values such as the size of the observation and action vectors.
+env = LyceumMuJoCo.HopperV2();
+dobs, dact = length(obsspace(env)), length(actionspace(env));
+
+#md # ## Policy Gradient Components
+# Policy Gradient methods require a policy: a function that takes in the state/observations
+# of the agent, and output an action i.e. `action = π(obs)`.
+# In much of Deep RL, the policy takes the form of a neural network which can be built
+# on top of the [Flux.jl](https://github.com/FluxML/Flux.jl) library.
+# The network below is two layers, mapping from our observation space to 32 hidden units,
+# to the second 32 hidden units layer, before emitting a vector of actions. The activations
+# are all tanh functions, and we initialize the network with Glorot Uniform initializations.
+# The policy is more than just a feed forward nerual network, however. It's treated as a
+# stochastic variable, and thus we track the log of the standard deviation of noise to apply
+# to the action sampling; this is final zero vector of size 'dact'.
+policy = DiagGaussianPolicy(
+    multilayer_perceptron(
+        dobs,
+        32,
+        32,
+        dact;
+        σ = tanh,
+        initb = glorot_uniform,
+        initb_final = glorot_uniform,
+    ),
+    zeros(dact),
+)
+# Convert the weights of our policy to `Float32` rather than the default
+# Float64 for performance.
+policy = Flux.paramtype(Float32, policy);
+
+
+
+# This NPG implementation uses [Generalized Advantaged Estimation](https://arxiv.org/pdf/1506.02438.pdf),
+# which requires an estimate of the value function `V(state)`, which we represent using
+# a 2-layer, feedforward neural network.
+value = multilayer_perceptron(
+    dobs,
+    128,
+    128,
+    1;
+    σ = Flux.relu,
+    initb = glorot_uniform,
+    initb_final = glorot_uniform,
+)
+value = Flux.paramtype(Float32, value);
+
+
+
+
+# FluxTrainer is an iterator that loops on the Flux object provided.
+# The result at each loop is passed to stopcb below, so you can quit after
+# a number of epochs, convergence, or other criteria; here it's capped at two epochs
+# as a lambda function.
+valueloss(bl, X, Y) = Flux.mse(vec(bl(X)), vec(Y))
+valuetrainer = FluxTrainer(
+    optimiser = ADAM(1e-3),
+    szbatch = 64,
+    lossfn = valueloss,
+    stopcb = s -> s.nepochs > 2,
+);
+
+
+# The `NaturalPolicyGradient` iterator is a type that pre-allocates all necesary
```

Comment:
So I think once we have docstrings for e.g. `NaturalPolicyGradient` we can just skip having to discuss what each of the parameters do and link to the docs. Otherwise we'll just be repeating ourselves every single example/tutorial.